pallets/parachain-staking/src/migrations.rs

```diff
@@ -16,184 +16,34 @@
-
-		// Read old round info
-		let mut round: RoundInfo<BlockNumberFor<T>> = if let Some(bytes) =
-			unhashed::get_raw(&raw_key)
-		{
-			let len = bytes.len();
-			match len {
-				// Migration already done
-				20 => {
-					log::info!("MigrateRoundWithFirstSlot already applied.");
-					return Default::default();
-				}
-				// Migrate from rt2800
-				16 => match RoundInfoRt2800::decode(&mut &bytes[..]) {
-					Ok(round) => round.into(),
-					Err(e) => panic!("corrupted storage: fail to decode RoundInfoRt2800: {}", e),
-				},
-				// Migrate from rt2700
-				12 => match RoundInfoRt2700::decode(&mut &bytes[..]) {
-					Ok(round) => round.into(),
-					Err(e) => panic!("corrupted storage: fail to decode RoundInfoRt2700: {}", e),
-				},
-				// Storage corrupted
-				x => panic!(
-					"corrupted storage: parachainStaking.Round invalid length: {} bytes",
-					x
-				),
-			}
-		} else {
-			panic!("corrupted storage: parachainStaking.Round don't exist");
-		};
-
-		// Compute new field `first_slot``
-		round.first_slot = compute_theoretical_first_slot(
-			<frame_system::Pallet<T>>::block_number(),
-			round.first,
-			u64::from(T::SlotProvider::get()),
-			T::BlockTime::get(),
-		);
-
-		// Fill DelayedPayouts for rounds N and N-1
-		if let Some(delayed_payout) =
-			<crate::DelayedPayouts<T>>::get(round.current.saturating_sub(2))
-		{
-			<crate::DelayedPayouts<T>>::insert(
-				round.current.saturating_sub(1),
-				delayed_payout.clone(),
-			);
-			<crate::DelayedPayouts<T>>::insert(round.current, delayed_payout);
-		}
-
-		// Apply the migration (write new Round value)
-		crate::Round::<T>::put(round);
-
-		Default::default()
-	}
-
-	#[cfg(feature = "try-runtime")]
-	fn post_upgrade(_state: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
-		let _round = crate::Round::<T>::get(); // Should panic if SCALE decode fail
-		Ok(())
-	}
-}
-
-fn compute_theoretical_first_slot<BlockNumber: Saturating + Into<u64>>(
-	current_block: BlockNumber,
-	first_block: BlockNumber,
-	current_slot: u64,
-	block_time: u64,
-) -> u64 {
-	let blocks_since_first: u64 = (current_block.saturating_sub(first_block)).into();
-	let slots_since_first = match block_time {
-		12_000 => blocks_since_first * 2,
-		6_000 => blocks_since_first,
-		_ => panic!("Unsupported BlockTime"),
-	};
-	current_slot.saturating_sub(slots_since_first)
-}
-
 #[cfg(test)]
 mod tests {
 	use super::*;
+	use sp_runtime::Saturating;
+
+	use frame_support::traits::Saturating;
+	fn compute_theoretical_first_slot<BlockNumber: Saturating + Into<u64>>(
+		current_block: BlockNumber,
+		first_block: BlockNumber,
+		current_slot: u64,
+		block_time: u64,
+	) -> u64 {
+		let blocks_since_first: u64 = (current_block.saturating_sub(first_block)).into();
+		let slots_since_first = match block_time {
+			12_000 => blocks_since_first * 2,
+			6_000 => blocks_since_first,
+			_ => panic!("Unsupported BlockTime"),
+		};
+		current_slot.saturating_sub(slots_since_first)
+	}
```

Comment:
This can be removed, I do not think it is being used anymore

Reply:
Sure removed the code + the test in the migration as it is dependent on it in https://github.com/moonbeam-foundation/moonbeam/pull/2885/commits/a9f0e756af1ad0a0170f187a09b0a6cd1d059a63